airflow/contrib/operators/gcs_to_gcs.py

```diff
@@ -90,18 +97,20 @@ def execute(self, context):
             delegate_to=self.delegate_to
         )
 
+        copied_objects = []
         if '*' in self.source_object:
             wildcard_position = self.source_object.index('*')
             objects = hook.list(self.source_bucket,
                                 prefix=self.source_object[:wildcard_position],
                                 delimiter=self.source_object[wildcard_position + 1:])
 
             for source_object in objects:
+                destination_object_path = source_object[wildcard_position + 1:]
```

Comment:
Did you test this?

The `source_object` here would be the result of `hook.list` method above. Hence the path would be for example `dir1/file.csv` when the wildcard would probably have been used like `dir1/*.csv`. In that case `destination_object_path` should be `source_object[wildcard_position:]`. 

Correct me if I am wrong.

Reply:
You are correct, and the problem is in the code which adds a `/` character, which is not necessarily the intent here; I just blindly copied that from the previous code, and did not cover all the cases in tests. This needs fixing, but not in this PR; see below my response to @bolkedebruin